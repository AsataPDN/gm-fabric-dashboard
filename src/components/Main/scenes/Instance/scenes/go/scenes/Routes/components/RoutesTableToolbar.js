import { PropTypes } from "prop-types";
import React from "react";
import GMSelect from "../../../../../../../components/GMSelect";
import { ButtonSecondaryText } from "../../../../../../../../GMButtons";

RoutesTableToolbar.propTypes = {
  filterString: PropTypes.string.isRequired,
  keyToSortBy: PropTypes.string.isRequired,
  setFilterString: PropTypes.func.isRequired,
  setKeyToSortBy: PropTypes.func.isRequired
};
/**
 * UI controls for filtering and sorting the Go RoutesTable
 * @export
 * @param {any} {
 *   filterString,
 *   setFilterString,
 *   setKeyToSortBy
 * }
 * @returns JSX.Element
 */
export default function RoutesTableToolbar({
  filterString,
  keyToSortBy,
  setFilterString,
  setKeyToSortBy
}) {
  return (
    <div className="toolbar">
      <div className="toolbar-left">
        <form>
          <input
            className="form-control"
            onChange={evt => setFilterString(evt.target.value)}
            placeholder="Search Routes"
            aria-label="Search All Routes"
            type="search"
            value={filterString}
          />
        </form>
      </div>
      <div className="toolbar-right">
        <GMSelect
          name="form-field-group-by"
          options={[
            {
              value: "route",
              label: "Route"
            },
            {
              value: "requests",
              label: "Requests"
            },
            {
              value: "errorCount",
              label: "Error Count"
            },
            {
              value: "latency50",
              label: "Latency 50%"
            },
            {
              value: "latency99",
              label: "Latency 99%"
            }
          ]}
          value={keyToSortBy}
          onChange={val => setKeyToSortBy(val.value)}
          clearable={false}
          searchable={false}
          valueRenderer={val => [
            <span>
              <span>Sort </span>
              <ButtonSecondaryText>{val.label}</ButtonSecondaryText>
            </span>
          ]}
        />
      </div>
    </div>
  );
}
