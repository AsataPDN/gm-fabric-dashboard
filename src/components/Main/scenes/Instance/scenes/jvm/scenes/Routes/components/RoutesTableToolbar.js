import { PropTypes } from "prop-types";
import React from "react";

import GMSelect from "components/Main/components/GMSelect";
import SecondaryText from "components/SecondaryText";
import Toolbar from "components/Main/components/Toolbar";
import ToolbarLeft from "components/Main/components/ToolbarLeft";
import ToolbarRight from "components/Main/components/ToolbarRight";
import SearchInput from "components/Main/components/GMSearchInput";

RoutesTableToolbar.propTypes = {
  filterString: PropTypes.string.isRequired,
  keyToSortBy: PropTypes.string.isRequired,
  setFilterString: PropTypes.func.isRequired,
  setKeyToSortBy: PropTypes.func.isRequired
};
/**
 * UI controls for filtering and sorting the RoutesTable
 * @export
 * @param {any} {
 *   filterString,
 *   setFilterString,
 *   setKeyToSortBy
 * }
 * @returns JSX.Element
 */
export default function RoutesTableToolbar({
  filterString,
  keyToSortBy,
  setFilterString,
  setKeyToSortBy
}) {
  return (
    <Toolbar>
      <ToolbarLeft>
        <form>
          <SearchInput
            onChange={evt => setFilterString(evt.target.value)}
            placeholder="Search Routes"
            aria-label="Search All Routes"
            value={filterString}
          />
        </form>
      </ToolbarLeft>
      <ToolbarRight>
        <GMSelect
          name="form-field-group-by"
          options={[
            {
              value: "route",
              label: "Route"
            },
            {
              value: "totalRequests",
              label: "Total Requests"
            },
            {
              value: "errorPercent",
              label: "Error %"
            }
          ]}
          value={keyToSortBy}
          onChange={val => setKeyToSortBy(val.value)}
          clearable={false}
          searchable={false}
          valueRenderer={val => (
            <span>
              <span>Sort </span>
              <SecondaryText>{val.label}</SecondaryText>
            </span>
          )}
        />
      </ToolbarRight>
    </Toolbar>
  );
}
