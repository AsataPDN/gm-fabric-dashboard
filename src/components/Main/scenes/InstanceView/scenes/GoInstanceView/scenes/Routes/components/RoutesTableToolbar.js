import { PropTypes } from "prop-types";
import React from "react";

import GMSelect from "components/Main/components/GMSelect";
import Toolbar from "components/Main/components/Toolbar";
import ToolbarLeft from "components/Main/components/ToolbarLeft";
import ToolbarRight from "components/Main/components/ToolbarRight";
import SecondaryText from "components/SecondaryText";
import SearchInput from "components/Main/components/GMSearchInput";

RoutesTableToolbar.propTypes = {
  filterString: PropTypes.string.isRequired,
  keyToSortBy: PropTypes.string.isRequired,
  setFilterString: PropTypes.func.isRequired,
  setKeyToSortBy: PropTypes.func.isRequired
};
/**
 * UI controls for filtering and sorting the Go RoutesTable
 * @export
 * @param {any} {
 *   filterString,
 *   setFilterString,
 *   setKeyToSortBy
 * }
 * @returns JSX.Element
 */
export default function RoutesTableToolbar({
  filterString,
  keyToSortBy,
  setFilterString,
  setKeyToSortBy
}) {
  return (
    <Toolbar>
      <ToolbarLeft>
        <form>
          <SearchInput
            onChange={evt => setFilterString(evt.target.value)}
            placeholder="Search Routes"
            aria-label="Search All Routes"
            value={filterString}
          />
        </form>
      </ToolbarLeft>
      <ToolbarRight>
        <GMSelect
          name="form-field-group-by"
          options={[
            {
              value: "route",
              label: "Route"
            },
            {
              value: "requests",
              label: "Requests"
            },
            {
              value: "errorCount",
              label: "Error %"
            },
            {
              value: "latency50",
              label: "Latency 50%"
            },
            {
              value: "latency99",
              label: "Latency 99%"
            }
          ]}
          value={keyToSortBy}
          onChange={val => setKeyToSortBy(val.value)}
          clearable={false}
          searchable={false}
          valueRenderer={val => [
            <span key={val}>
              <span>Sort </span>
              <SecondaryText>{val.label}</SecondaryText>
            </span>
          ]}
        />
      </ToolbarRight>
    </Toolbar>
  );
}
