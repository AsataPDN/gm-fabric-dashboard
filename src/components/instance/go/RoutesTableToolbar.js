import { PropTypes } from "prop-types";
import React from "react";

const spanWidth = {
  flexBasis: "2em"
};

RoutesTableToolbar.propTypes = {
  filterString: PropTypes.string.isRequired,
  setFilterString: PropTypes.func.isRequired,
  setKeyToSortBy: PropTypes.func.isRequired
};
/**
 * UI controls for filtering and sorting the Go RoutesTable
 * @export
 * @param {any} {
 *   filterString,
 *   setFilterString,
 *   setKeyToSortBy
 * }
 * @returns JSX.Element
 */
export default function RoutesTableToolbar({
  filterString,
  setFilterString,
  setKeyToSortBy
}) {
  return (
    <div className="toolbar">
      <div className="toolbar-left">
        <form>
          <input
            className="form-control"
            onChange={evt => setFilterString(evt.target.value)}
            placeholder="Search Routes"
            type="search"
            value={filterString}
          />
        </form>
      </div>
      <div className="toolbar-right">
        <div className="uk-button-group">
          <button className="btn">
            <span className="label" style={spanWidth}>
              Sort
            </span>
            <span
              className="icon"
              data-uk-icon="icon: triangle-down"
              style={spanWidth}
            />
          </button>
          <div data-uk-dropdown="mode: click; pos: bottom-right; boundary: ! .uk-button-group; boundary-align: true;">
            <ul className="uk-nav uk-dropdown-nav">
              <li onClick={evt => setKeyToSortBy("route")}>Route</li>
              <li onClick={evt => setKeyToSortBy("requests")}>Requests</li>
              <li onClick={evt => setKeyToSortBy("errorCount")}>Error Count</li>
              <li onClick={evt => setKeyToSortBy("latency50")}>Latency 50%</li>
              <li onClick={evt => setKeyToSortBy("latency99")}>Latency 99%</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
}
