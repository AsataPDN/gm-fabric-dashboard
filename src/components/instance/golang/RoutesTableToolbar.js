import { PropTypes } from "prop-types";
import React from "react";

RoutesTableToolbar.propTypes = {
  filterString: PropTypes.string.isRequired,
  setFilterString: PropTypes.func.isRequired,
  setKeyToSortBy: PropTypes.func.isRequired
};
/**
 * UI controls for filtering and sorting the Golang RoutesTable
 * @export
 * @param {any} {
 *   filterString,
 *   setFilterString,
 *   setKeyToSortBy
 * } 
 * @returns JSX.Element
 */
export default function RoutesTableToolbar({
  filterString,
  setFilterString,
  setKeyToSortBy
}) {
  return (
    <div className="toolbar">
      <div className="toolbar-left">
        <form>
          <input
            className="form-control"
            onChange={evt => setFilterString(evt.target.value)}
            placeholder="Search Routes"
            type="search"
            value={filterString}
          />
        </form>
      </div>
      <div className="toolbar-right">
        <div className="uk-button-group">
          <button className="btn">
            <span className="label">Sort</span>
            <span className="icon" data-uk-icon="icon: triangle-down" />
          </button>
          <div data-uk-dropdown="mode: click; pos: bottom-right; boundary: ! .uk-button-group; boundary-align: true;">
            <ul className="uk-nav uk-dropdown-nav">
              <li onClick={evt => setKeyToSortBy("route")}>Route</li>
              <li onClick={evt => setKeyToSortBy("requests")}>Requests</li>
              <li onClick={evt => setKeyToSortBy("errorCount")}>Error Count</li>
              <li onClick={evt => setKeyToSortBy("latency50")}>Latency 50%</li>
              <li onClick={evt => setKeyToSortBy("latency99")}>Latency 99%</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
}
