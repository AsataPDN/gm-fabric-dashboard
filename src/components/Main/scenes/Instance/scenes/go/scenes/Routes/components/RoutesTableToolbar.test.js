import React from "react";
import { mount, render } from "enzyme";
import renderer from "react-test-renderer";

import RoutesTableToolbar from "./RoutesTableToolbar";

const RoutesTableToolbarWithProps = (
  <RoutesTableToolbar
    filterString={"topics"}
    keyToSortBy={"route"}
    setFilterString={jest.fn(string =>
      console.log(`setFilterString called with: ${string}`)
    )}
    setKeyToSortBy={jest.fn(() => console.log("setKeyToSortBy"))}
  />
);

describe("Go Instance Routes View: <RoutesTableToolbar />", () => {
  test("matches snapshot", () => {
    const tree = renderer.create(RoutesTableToolbarWithProps).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("calls setFilterString on input change", () => {
    let wrapper = mount(RoutesTableToolbarWithProps);
    wrapper
      .find({ type: "search" })
      .simulate("change", { target: { value: "category" } });
    expect(wrapper.props().setFilterString).toHaveBeenCalled();
  });

  // simulate dropdown selection change using onChange
  test("calls setKeyToSortBy on dropdown selection change", () => {
    let wrapper = mount(RoutesTableToolbarWithProps);

    wrapper
      .find("Select")
      .props()
      .onChange("latency50");
    expect(wrapper.props().setKeyToSortBy).toHaveBeenCalled();
  });
});
