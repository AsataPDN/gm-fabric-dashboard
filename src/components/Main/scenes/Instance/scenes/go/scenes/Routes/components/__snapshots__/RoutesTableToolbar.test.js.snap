// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Go Instance Routes View: <RoutesTableToolbar /> matches snapshot 1`] = `
<div
  className="sc-bwzfXH jrpUDe"
>
  <div
    className="sc-htpNat jnCRLF"
  >
    <form>
      <input
        aria-label="Search All Routes"
        className="form-control"
        onChange={[Function]}
        placeholder="Search Routes"
        type="search"
        value="topics"
      />
    </form>
  </div>
  <div
    className="sc-bxivhb WlBFC"
  >
    <div
      className="Select sc-bdVaJa bdoijW Select--single has-value"
      style={undefined}
    >
      <input
        disabled={false}
        name="form-field-group-by"
        type="hidden"
        value="route"
      />
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-value"
            style={undefined}
            title={undefined}
          >
            <span
              aria-selected="true"
              className="Select-value-label"
              id="react-select-2--value-item"
              role="option"
            >
              <span>
                <span>
                  Sort 
                </span>
                <span
                  className="sc-ifAKCX dAlnhc"
                >
                  Route
                </span>
              </span>
            </span>
          </div>
          <div
            aria-activedescendant="react-select-2--value"
            aria-expanded={false}
            aria-owns=""
            aria-readonly="false"
            className="Select-input"
            onBlur={[Function]}
            onFocus={[Function]}
            role="combobox"
            style={
              Object {
                "border": 0,
                "display": "inline-block",
                "width": 1,
              }
            }
            tabIndex={0}
          />
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
